src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -88,14 +102,32 @@ void PairingSession::DiscardExchange()
     }
 }
 
-CHIP_ERROR PairingSession::EncodeMRPParameters(TLV::Tag tag, const ReliableMessageProtocolConfig & mrpLocalConfig,
-                                               TLV::TLVWriter & tlvWriter)
+CHIP_ERROR PairingSession::EncodeSessionParameters(TLV::Tag tag, const Optional<ReliableMessageProtocolConfig> & providedMrpConfig,
+                                                   TLV::TLVWriter & tlvWriter)
 {
+    ReliableMessageProtocolConfig mrpLocalConfig = GetDefaultMRPConfig();
+    if (providedMrpConfig.HasValue())
```

Comment:
And if not, why would we send default values that may or may not have anything to do with reality?  Shouldn't it be an error to not have MRP parameters here?

Followup for sorting that out, please, but this is not passing the smell test for me.

Reply:
Previously if a node didn't provide have this local config, the other side would assign `mRemoteMRPConfig` to GetDefaultMRPConfig() which is what would ultimately get used.

https://github.com/project-chip/connectedhomeip/blob/master/src/protocols/secure_channel/PairingSession.h#L236

I can open up a follow up issue if you think that the old behaviour was broken and only now it seems to smell off

Reply:
Also to be clear, the reason why I am sending anything is because spec says:

`For backwards compatibility, if any tag after tag 2 (SESSION_ACTIVE_INTERVAL) is present, then the SESSION_ACTIVE_INTERVAL SHALL also be present.`

I could only set SESSION_ACTIVE_INTERVAL to the default if we are sending anything with a tag after 2. Please let me know if following that exact text is what I should be doing here

Reply:
We always send things with a tag after 2, right?  So we have to do something here to send the MRP values.

I think a bigger issue is whether GetDefaultMRPConfig() is the same on both sides of the communication.  Is it?

And more importantly: under what conditions is this HasValue() false?

Again, all of this can be a followup.

Reply:
Added a todo, which is tracked as an issue (https://github.com/project-chip/connectedhomeip/issues/30456). This is also tracked in my list of things to take care of as a part of batch commands